src/User/Keycloak/CachedUserIdentityResolver.php

```diff
@@ -25,38 +26,68 @@ public function __construct(
 
     public function getUserById(string $userId): ?UserIdentityDetails
     {
-        return $this->deserializeUserIdentityDetails(
-            $this->cache->get(
-                $this->createCacheKey($userId, 'user_id'),
-                function () use ($userId) {
-                    return $this->getUserIdentityDetailsAsArray($this->userIdentityResolver->getUserById($userId));
-                }
-            )
-        );
+        try {
```

Comment:
This function is becoming rather complex. This is something that can/should be tackled when we are on PHP 8

Reply:
Try to throw the `CacheException` inside the `getUserIdentityDetailsAsArray` method and use an arrow function.